modules/pam_faillock/pam_faillock.c

```diff
@@ -192,6 +192,13 @@ read_config_file(pam_handle_t *pamh, struct options *opts, const char *cfgfile)
 	char linebuf[FAILLOCK_CONF_MAX_LINELEN+1];
 
 	f = fopen(cfgfile, "r");
+#ifdef VENDOR_FAILLOCK_DEFAULT_CONF
+	if (f == NULL && errno == ENOENT && cfgfile == default_faillock_conf) {
+	  /* If the specified default file in /etc does not exist,
+             try the vendor file as fallback. */
+	  f = fopen(VENDOR_FAILLOCK_DEFAULT_CONF, "r");
+	}
+#endif
 	if (f == NULL) {
 		/* ignore non-existent default config file */
 		if (errno == ENOENT && cfgfile == default_faillock_conf)
```

Comment:
Before this change, `read_config_file` used to return `PAM_SUCCESS` when `fopen(default_faillock_conf, "r")` failed with `ENOENT`.
After this change, `read_config_file` would return `PAM_SERVICE_ERR` when attempts of opening `default_faillock_conf` and `VENDOR_FAILLOCK_DEFAULT_CONF` are failed with `ENOENT`.

I bet this change of behaviour was not intended.